synapse/groups/groups_server.py

```diff
@@ -205,6 +206,24 @@ def delete_group_summary_room(self, group_id, requester_user_id,
 
         defer.returnValue({})
 
+    @defer.inlineCallbacks
+    def set_group_joinable(self, group_id, requester_user_id, content):
+        """Sets whether a group is joinable without an invite or knock
+        """
+        yield self.check_group_is_ours(
+            group_id, requester_user_id, and_exists=True, and_is_admin=requester_user_id
+        )
+
+        is_joinable = content.get('joinable')
+        if is_joinable is None:
```

Comment:
Shouldn't we also check if is_joinable evaluates to a boolean?
Or is it enough to handle that on a database level as storing arbitrary data into a boolean field will fail?